myning/utils/race_rarity.py

```diff
@@ -38,8 +38,33 @@
 RACE_WEIGHTS = [150, 75, 40, 20, 10, 7, 4]
 
 
-def get_recruit_species(highest_rarity: int):
+def _recruit_in_tier(tier: list[CharacterRaces]) -> CharacterRaces:
+    if len(tier) == 0:
+        raise ValueError("Empty tier")
     player = Player()
+    facility = ResearchFacility()
+    undiscovered = [
+        RACES[race_index] for race_index in tier if RACES[race_index] not in player.discovered_races
+    ]
+    undiscovered_probability = len(undiscovered) / len(tier)
+    percent_increase = 0
+    if facility.has_research("species_discovery"):
+        percent_increase = RESEARCH["species_discovery"].player_value / 100
+    undiscovered_probability += undiscovered_probability * percent_increase
+    discovered_probability = 1 - undiscovered_probability
+    undiscovered_weight = undiscovered_probability / len(undiscovered)
+    discovered_weight = discovered_probability / (len(tier) - len(undiscovered))
```

Comment:
A little fuzzy on this section. Will this be fine if they haven't researched any? Will it work if they have unlocked all races in the tiers?

Reply:
percent_increase will be zero if no research has happened, so no research is fine. It will just be default weights for everything in the tier. If they have unlocked everything, a division by zero error would occur. Good catch lol.